# BIOEN 455/556 Winter 2026 Lab 4: Domino Valves & Capillaric Circuits 
**By Sam Koplik and Ayo Olanrewaju (Feb 2025), Modified By Aileen Sun (Feb 2026)**  

---

## Objective  
To demonstrate the function of domino valves in CCs by observing sequential liquid delivery, measuring reservoir drainage times, and comparing experimental results to theoretical flow rates and capillary pressures obtained using python. **Based on the paper: [Leong, K. M., et al. (2024). Democratizing Access to Microfluidics: Rapid Prototyping of Open Microchannels with Low-Cost LCD 3D Printers.ACS Omega](https://pubs.acs.org/doi/full/10.1021/acsomega.4c07776)**  


---

## Instructions/ Submission Guidelines 
- Complete any section of code marked with a `TO-DO:`.
- Save your work frequently to prevent data loss.
- <font color="red">Answer reflection questions (in red).</font> This can be completed after lab. You can type this in a markdown within this script. You may optionally take videos or pictures your the chip to include in your reflection. 
  
  

**For Submission:**
1. Add your name and the date to the designated cell in the notebook.  
2. Once finished, download and submit your modified script and reflection answers to Canvas as a **HTML** file and submit the **.ipynb** file:  
   - Go to **File > Save and Export Notebook As > HTML**.
   - Go to **File > Download** to save the ipynb file.

---

## Name: Sarah Salarda
## Date: 02/10/2026
---

# <font color='blue'> Part 1: Calculating Theoretical Pressure and Flow Resistance for CC Components</font>

## Watch the video (below) from the paper [Leong, K. M., et al. (2024). Democratizing Access to Microfluidics: Rapid Prototyping of Open Microchannels with Low-Cost LCD 3D Printers. ACS Omega](https://pubs.acs.org/doi/full/10.1021/acsomega.4c07776) to orient yourself with the capillaric circuit that we will be using for today's lab.

In [2]:
# from IPython.display import Video
# Video("lab4_images/ao4c07776_si_001.mp4", width=900)

## Recall from Lab 2  
The following functions are carried forward from Lab 2 and 3 and will be used in this lab:  
- **`flowResistance`**: Calculates the resistance to flow in a microchannel.  
- **`capPressure`**: Determines the capillary pressure based on contact angles and channel dimensions.

Be sure you understand the principles and formulas behind these functions, as they will be applied in the analysis for this lab.

You **do not need to edit** this next cell, just run it.


### Constants and Assumptions
#### The following constants are assumed for the capillary pressure calculation:
- #### Contact angle hysteresis is an important consideration when designing RBVs, as advancing contact angles are critical for channel filling while receding contact angles are essential for channel draining.

- #### Advancing and Receding Contact Angles:

    - **Top Receding Contact Angle (θTopReceding)**: A fixed value representing the contact angle at the top of the microchannel when the fluid is receding.  
      `θTopReceding = (np.pi * 89/180)`
    
    - **Top Advancing Contact Angle (θTopAdvancing)**: A fixed value representing the contact angle at the top of the microchannel when the fluid is advancing.  
      `θTopAdvancing = (np.pi * 114/180)`
    
    - **Bottom Receding Contact Angle (θBottomReceding)**: A fixed value representing the contact angle at the bottom of the microchannel when the fluid is receding.  
      `θBottomReceding = (np.pi * 31/180)`
    
    - **Bottom Advancing Contact Angle (θBottomAdvancing)**: A fixed value representing the contact angle at the bottom of the microchannel when the fluid is advancing.  
      `θBottomAdvancing = (np.pi * 45/180)`

In [3]:
#just run this cell don't edit. This defines the functions we are using that we wrote in lab 2.
import numpy as np

### CONSTANTS - in SI units
#global variables
# Properties of liquid
viscosity = 8.9e-4  #Pa*s

# Surface tension of water is 0.073
surfTension = 0.069  #N/meter

#contact angle measurement, in radians
#hydrophobic top PDMS surface and hydrophilic side and bottom surfaces respectively
thetaTopWall = (np.pi * 89/180)     
thetaSideWalls = (np.pi * 31/180)


# FLOWRES
def flowResistance(w,h,l):
    h_actual = min(w,h)
    w_actual= max(w,h)
    flowRes = ((12*viscosity*l)/(1-0.63*(h_actual/w_actual)))*(1/(w_actual*(h_actual**3)))
    return flowRes

# CAPPRESSURE
def capPressure(w, h, thetaTopWall, thetaSideWalls):
    #Capillary pressure calculation for wetting liquid
    h_actual = min(w,h)
    w_actual= max(w,h)
    capPressure =surfTension*((np.cos(thetaTopWall)/h_actual) +(np.cos(thetaSideWalls)/h_actual) +(2 * np.cos(thetaSideWalls)/w_actual))
    return capPressure

## Calculate the capPressure and flowResistance for the RBV (attached to reservoirs) during draining. Use the `receding contact angle` for draining channel components.

In [4]:
thetaBottomReceding = (np.pi * 31/180)
thetaTopReceding = (np.pi * 89/180)

#dimensions of RBV (in branch 1)
w1 = 1513.6e-6  #meters    
h1 = 600e-6 #meters 
l1 = 4266e-6 #meters 

#dimensions of each RBV (in branches 2-4)
w234 = 172e-6  #meters    
h234 = 600e-6 #meters 
l234 = 3508.8e-6 #meters 

#TO-DO: Call and print capPressure and flowResistance for the 4 RBVs. Use the receding contact angle. (0.5 pts)
capPressure01 = capPressure(w1, h1, thetaTopReceding, thetaBottomReceding)
flowResistance01 = flowResistance(w1,h1,l1)
print("Capillary pressure for RBV in branch 1", capPressure01)
print("Flow resistance for RBV in branch 1", flowResistance01)

capPressure234 = capPressure(w234, h234,thetaTopReceding, thetaBottomReceding)
flowResistance234 = flowResistance(w234, h234, l234)
print("Capillary Pressure for RBV in branch 2-4", capPressure234)
print("Flow resistance for RBV in branch 2-4", flowResistance234)

Capillary pressure for RBV in branch 1 178.73209051295146
Flow resistance for RBV in branch 1 185743219.44346604
Capillary Pressure for RBV in branch 2-4 548.0133616767087
Flow resistance for RBV in branch 2-4 14979499965.216139


### **<font color="red"> Reflection Question 1: (0.2 pts)</font>**
-  ### **<font color="red"> Based on the RBV calculations, which reservoir will drain first?**

### TO-DO: Branch 1 will be the first to drain because it has the smallest capillary pressure compared to all other RBVs.

## Calculate the capPressure and flowResistance for the TVs (attached to reservoirs) during draining. Use the `receding contact angle` for draining channel components.

In [5]:
#dimensions of each TV
#TVs
w_TV = 206.4e-6  #meters    
h_TV = 140e-6 #meters 
l_TV = 2304e-6 #meters 

#TO-DO: Call and print capPressure and flowResistance for the TVs. Use the receding contact angle. (0.5 pts)
capPressureTV = capPressure(w_TV, h_TV, thetaTopReceding, thetaBottomReceding)
flowResistanceTV = flowResistance(w_TV, h_TV, l_TV)
print("Capillary pressure for each TVs", capPressureTV)
print("Flow Resistance of each TV", flowResistanceTV)

Capillary pressure for each TVs 1004.1686140213604
Flow Resistance of each TV 75866865963.20909


## We will also calculate the resistance and capillary pressure of the pump/main resistor. We will use the `advancing contact angle` for this.

In [6]:
thetaBottomAdvancing = (np.pi * 45/180)
thetaTopAdvancing = (np.pi * 114/180)
#dimensions of Pump resistor
#Pump
w_Pump = 200e-6  #meters    
h_Pump = 200e-6 #meters 
l_Pump = 26000e-6 #meters 

#TO-DO: Call and print capPressure and flowResistance for the Pump. Use the ADVANCING contact angle. (0.5 pts)
capPressurePump = capPressure(w_Pump, h_Pump, thetaTopAdvancing, thetaBottomAdvancing)
flowResistancePump = flowResistance(w_Pump, h_Pump, l_Pump)
print("Capillary pressure for the pump/main resistor", capPressurePump)
print("Flow Resistance of each pump/main resistor", flowResistancePump)

Capillary pressure for the pump/main resistor 591.5313766669257
Flow Resistance of each pump/main resistor 469054054054.05383


## Calculate the capPressure, flowResistance for the 4 Reservoirs.  Use the `receding contact angle` for draining channel components.


In [7]:
#Reservoirs 1-4
w_Res = 1513.6e-6  #meters    
h_Res = 600e-6 #meters 
l_Res = 8496.8e-6 #meters 

#TO-DO: Call and print capPressure, flowResistance for each reservoir. Use the receding contact angle. (0.5 pts)
capPressureRes = capPressure(w_Pump, h_Pump, thetaTopReceding, thetaBottomReceding)
flowResistanceRes = flowResistance(w_Res, h_Res, l_Res)
print("Capillary pressure for each Reservoir (1-4)", capPressureRes)
print("Flow Resistance of each Reservoir (1-4)", flowResistanceRes)

Capillary pressure for each Reservoir (1-4) 893.1892364475492
Flow Resistance of each Reservoir (1-4) 369953817.8544872


## Next we will calculate the capillary pressure of the new channel that forms when the air conduit is open (the new RBV that forms when the air conduit is open). This is depicted in the #4 image below.


<div style="text-align: center;">
  <img src="lab4_images/partb.jpeg" width="1000" />
</div>





In [8]:
#domino valve dimensions
w_DV = 997.6e-6  #meters    
h_DV = 1000e-6 #meters 
l_DV = 8596.8e-6 #meters 

#TO-DO: Call and print capPressure for the domino valve. Use the receding contact angle. (0.5 pts)
capPressureDV = capPressure(w_DV, h_DV, thetaTopReceding, thetaBottomReceding)
print("Capillary pressure for each DV", capPressureDV)

Capillary pressure for each DV 178.7830327581368


### **<font color="red"> Reflection Question 2: (0.8 pts)</font>** 
- ### **<font color="red"> How do these capillary pressure differences between RBVs during drainage (based on Leong et al 2024) compare with those in lab 3 (based on Olanrewaju et al 2016)? (0.2 pts)</font>**
- ### **<font color="red"> In the scenario where the domino valve (air conduit) is open, and reservoir 1 has drained, which reservoir drains next? Comment on the capillary pressures driving flow and the theoretical values. (0.2 pts)**
- ### **<font color="red"> Comment on how sequential drainage is achieved in Leong et al 2024 versus Olanrewaju et al 2016. (0.2 pts)</font>**
- ### **<font color="red">  Based on the video you watched at the beginning of lab, the control channel starts draining prematurely (30s mark of the video). Why do you think this happens? And how could it be avoided? (0.2 pts)</font>**

### TO-DO: The capillary pressure in lab 3 [-194, -271, -358, -441] follows an increasing pattern through each branch unlike here where its low for reservoir 1 but the same in reservoir 3-4. However, this does not change the drainage pattern where each branch releases sequentially. 

### If the domino valve (air conduit) is open and reservoir 1 (R1) is drained, the next reservoir should be reservoir 2 (R2) which is connected to reservoir 1 through an airlink. As reservoir 1 finishes draining, the air link from R1 to R2 causes a pressure drop within reservoir 2 that lets it drain immediately. Focusing on the domino valve capillary pressure we calculated (178.78 Pa), this value larger than reservoir 1 but smaller than reservoirs 2-4 so it can create that domino effect. 

### In Olanreqaju,sequential drainage is achieved through graded RBV capillary pressures where each reservoir has a higher RBV than the previous so it can release in that order. Leong uses domino valves that connect each reservoir and allows a domino affect once the upstream reservoir finishes draining. 

### The control channel starts draining prematurely because the control channel might have an RBV with a weak capillary barrier than what is intended so the pressure exceeds the burst pressure too early. A weak capillary could specifically be due to trapped air within the channel, imperfect surfacce treatment on the chip, or faulty dimensions when manufacturing the chip. 

# <font color='blue'> Part 2: Calculating Equivalent Resistance in Branches</font>

## Before we calculate equivalent resistances for each branch, we need to calculate the pressure and resistance of the portion of the control channel that connects the TVs to the pump/main resistor for each branch. Since the length of that control channel actively being used varies depending on which branch is draining. This will be unique to each branch.


<div style="text-align: center;">
  <img src="lab4_images/circuit.png" width="800" />
</div>



In [9]:
#control channel part 1
w_cc_p1 = 619.2e-6  #meters    
h_cc_p1 = 600e-6 #meters 
l_cc_p1 = 2810e-6 #meters 


#control channel part 2
w_cc_p2 = 619.2e-6  #meters    
h_cc_p2 = 600e-6 #meters 
l_cc_p2 = 8850e-6 #meters 


#control channel part 3
w_cc_p3 = 619.2e-6  #meters    
h_cc_p3 = 600e-6 #meters 
l_cc_p3 = 8759e-6 #meters 


#control channel part 4
w_cc_p4 = 619.2e-6  #meters    
h_cc_p4 = 600e-6 #meters 
l_cc_p4 = 8081e-6 #meters

#TO-DO calculate pressure and resistance for control channel parts 1-4. Use the receding contact angle. (0.5 pts)
capPressureCC1 = capPressure(w_cc_p1, h_cc_p1, thetaTopReceding, thetaBottomReceding)
flowResistanceCC1 = flowResistance(w_cc_p1, h_cc_p1, l_cc_p1)
print("Capillary pressure for control channel 1", capPressureCC1)
print("Flow Resistance for control channel 1", flowResistanceCC1)

capPressureCC2 = capPressure(w_cc_p2, h_cc_p2, thetaTopReceding, thetaBottomReceding)
flowResistanceCC2 = flowResistance(w_cc_p2, h_cc_p2, l_cc_p2)
print("Capillary pressure for control channel 2", capPressureCC2)
print("Flow Resistance for control channel 2", flowResistanceCC2)

capPressureCC3 = capPressure(w_cc_p3, h_cc_p3, thetaTopReceding, thetaBottomReceding)
flowResistanceCC3 = flowResistance(w_cc_p3, h_cc_p3, l_cc_p3)
print("Capillary pressure for control channel 3", capPressureCC3)
print("Flow Resistance for control channel 3", flowResistanceCC3)

capPressureCC4 = capPressure(w_cc_p4, h_cc_p4, thetaTopReceding, thetaBottomReceding)
flowResistanceCC4 = flowResistance(w_cc_p4, h_cc_p4, l_cc_p4)
print("Capillary pressure for control channel 4", capPressureCC4)
print("Flow Resistance for control channel 4", flowResistanceCC4)

Capillary pressure for control channel 1 291.61661434572613
Flow Resistance for control channel 1 576031877.6487932
Capillary pressure for control channel 2 291.61661434572613
Flow Resistance for control channel 2 1814192924.2675514
Capillary pressure for control channel 3 291.61661434572613
Flow Resistance for control channel 3 1795538511.147964
Capillary pressure for control channel 4 291.61661434572613
Flow Resistance for control channel 4 1656552883.7295008


## Using the equation for calculating equivalent resistance in series, you will calculate the equivalent resistance for each branch (while it is draining). To do this we will need to determine the path of flow and sum resistors in series along that path. 


<div style="text-align: center;">
  <img src="lab4_images/circuit.png" width="800" />
</div>


## When channels are connected end-to-end (in series), the total resistance to flow is the sum of the individual resistances where fluid is flowing. Keep in mind not every resistor shown in the branch has fluid flow. You should only include resistors where fluid flows during drainage:

$$
\displaystyle \huge R_{\text{eq, series}} = R_1 + R_2 + \dots + R_n
$$



In [10]:
#TO-DO calculate equivalent resistance for branches 1-4. You do not need to do the control channel branch. (0.5 pts)
R1 = flowResistanceTV + flowResistanceRes + flowResistance01
print("Equivalent resistane for branch 1", R1)

R2 = flowResistanceTV + flowResistanceRes + flowResistance234
print("Equivalent resistane for branch 1", R2)

R3 = flowResistanceTV + flowResistanceRes + flowResistance234
print("Equivalent resistane for branch 1", R3)

R4 = flowResistanceTV + flowResistanceRes + flowResistance234
print("Equivalent resistane for branch 1", R4)

Equivalent resistane for branch 1 76422563000.50705
Equivalent resistane for branch 1 91216319746.27972
Equivalent resistane for branch 1 91216319746.27972
Equivalent resistane for branch 1 91216319746.27972


In [ ]:
diff_R1 = (R1 + flowResistanceCC1) - R1
print("Difference in flow resistance with CC", diff_R1)
diff_R2 = (R2 + flowResistanceCC2) - R2
print("Difference in flow resistance with CC", diff_R2)
diff_R3 = (R3 + flowResistanceCC3) - R3
print("Difference in flow resistance with CC", diff_R3)
diff_R4 = (R4 + flowResistanceCC4) - R4
print("Difference in flow resistance with CC", diff_R4)

Difference in flow resistance with CC 576031877.6487885
Difference in flow resistance with CC 1814192924.2675476
Difference in flow resistance with CC 1795538511.1479645
Difference in flow resistance with CC 1656552883.7295074


### **<font color="red"> Reflection Question 3: (0.4 pts)</font>** 
- ### **<font color="red">How much of a difference in calculated flow resistance do we get if you ignore the resistance along the control channel (ignoring control channel parts)? (0.2 pts)</font>**
- ### **<font color="red">Why do you think this is the case? Comment on which CC elements impact the resistance the most. You do not need to redo the resistance calculations to answer this. You can just compare the component resistances and speculate based on that. (0.2 pts) </font>**

### TO-DO: If you ignore the resistance along the control channel, you lose 1x10^8 to 1x10^9 but in comparison to the overall resistance within each branch/reservoir that's at 1x10^10, this is actually a small amount.

### This is the case because the dominate resistors within this system are actually the trigger valves (around 7.5x10^10) which make up for a majoriy of the equivalent resistance within each branch and the RBVs (around 1.8x10^8 to 1.5x10^10). These large values together make the control channel's resistance negligible. 

# <font color='blue'> Part 3: Calculating Flow Rates in Branches</font>


## Now we will calculate flow rate in the reservoir. We will use **Ohm’s Law for Fluidic Circuits (Hagen-Poiseuille Analog)**  
#### In capillaric circuits, pressure difference drives fluid flow, just as voltage drives electrical current. The fluidic equivalent of Ohm’s Law can be expressed as: 

$$
\displaystyle \huge Q = \frac{\Delta P_{\text{}}}{R_{\text{total}}}
$$

Where:

- **Q** = flow rate (m³/s)
- **ΔP** =  (Pump pressure - RBV pressure). Keep in mind there is a **new RBV pressure** in channels 2-4 during draining.
- **R**<sub>total</sub> = total resistance in branch (Pa·s/m³)

## We will use ΔP = (the pressure the pump - the pressure of each RBV). We will also use the equivalent resistance of each branch to get the flow rate of liquid in each reservoir. Also, Make sure to convert your answer to µL/s so that we can easily compare the results to experimental flow rates and those reported in the paper.

In [12]:
#TO-DO calculate flow rates and convert flowRates to uL/s (they will be in m^3/s before you convert). (0.5 pts)
flowRate1 = (capPressurePump - capPressure01)/R1 * 10**9
print("Flow rate of branch 1", flowRate1) 

flowRate2 = (capPressurePump - capPressure234)/R2 * 10**9
print("Flow rate of branch 1", flowRate2) 

flowRate3 = (capPressurePump - capPressure234)/R3 * 10**9
print("Flow rate of branch 3", flowRate3) 

flowRate4 = (capPressurePump - capPressure234)/R4 * 10**9
print("Flow rate of branch 4", flowRate4) 

Flow rate of branch 1 5.401536796812682
Flow rate of branch 1 0.4770858450687702
Flow rate of branch 3 0.4770858450687702
Flow rate of branch 4 0.4770858450687702


# <font color='blue'> Part 4: Measuring drain time for domino valves  </font>

### Overview:
You will measure and record the drain time of liquids in a CC with domino valves and then calculate flow rates. The CC design is based on Figure 4 (below) from [Leong, K. M., et al. (2024). Democratizing Access to Microfluidics: Rapid Prototyping of Open Microchannels with Low-Cost LCD 3D Printers.ACS Omega](https://pubs.acs.org/doi/full/10.1021/acsomega.4c07776)


<div style="text-align: center;">
  <img src="lab4_images/images_large_ao4c07776_0004.jpeg" width="800" />
</div>



### Channel Dimensions

Here, we have provided a CAD screenshot for CC. You will measure and record drain time for each of the 4 reservoirs and subsequently the flow rates. We also have provided you space in a table to record drain times and flow rates.  We recommend also writing these measurements down outside of this notebook to avoid any saving issues and loss of data.

<div style="text-align: center;">
  <img src="lab4_images/CDV1L_v3.PNG" width="1000" />
</div>


### Materials
- Resin-printed microfluidic chip, plasma treated, and sealed with hydrophobic cover tape (ARseal 90697, Adhesives Research, Glen Rock, Pennsylvania, USA)
- Red and yellow dye solutions (13% food dye v/v in DI water). 
- P20 Micropipettes and tips
- Paper pump
- Kim wipes (for cleaning up)
- Gloves
- Timer or stopwatch

## Procedure (read fully before starting)

1. **Glove Up:** Wear gloves as part of standard lab safety and best practices. All the materials we’re working with, are not really hazardous. That said, it’s still best practice to wear gloves during handling—especially since we have no idea where these pipettes have been! 😊
2. **QC:**
   - Double-check that all channels are properly sealed with cover tapes and dye solution is mixed before starting.
   - Work in your petri dish and avoid tipping the chip (the dye may spill and stain whatever it touches)
3. **Filling Reservoirs:** Using a P20 pipette, introduce 11µL of the **red** dye solution into the inlet of reservoir 1. Introduce 9 µL of the **red** dye solution into the inlet of reservoirs 2-4.
4. **Filling Control Channel:** Using a  pipette, introduce 20µL of the **yellow** dye solution into the inlet of the control channel. Note that once you add liquid to the control channel, flow may begin in your first reservoir, so you should be ready with your timer!
5. **Paper Pump:** Attach a paper pump to the outlet of the CC if it is not done so already. You may need to put pressure on it so that it connects to the outlet and starts flow. If flow does not start or liquid does not fill the resistor fully, we can help you try and troubleshoot the chip!
6. **Timing:** Start the timer as soon as the red dye solution begins draining from each reservoir. We recommend using a lap function to more easily record the time stamps of sequential drainage. Alternatively, you can take a video with your phone of the drainage process and review the footage to obtain time stamps. If you go the video route, ensure you have enough storage to record a full video!
7. **Recording:** Record the time taken for each reservoir to drain. We suggest recording these times both in this notebook (in the table below) AND elsewhere to avoid any data loss. If your channel failed, note the modes of failure and mention this in your reflection question 3.
8. **Flow Rates:** Using the volumes provided (actual volumes in the reservoirs) and the measured drain times calculate the flow rates for each reservoir.


| Reservoir   | Drain Time (s) | Flow Rate (µL/s)  | Volume (µL) |
|-------------|----------------|-------------------|-------------|
| Reservoir 1 |        20        |       0.51            |    10.2     |
| Reservoir 2 |          17      |      0.49             |    8.4      |
| Reservoir 3 |            21   |        0.4           |    8.4      |
| Reservoir 4 |       37         |         0.23          |    8.4      |


### **<font color="red"> Reflection Question 4: (0.6 pts)</font>** 

- ### **<font color="red">How do the measured flow rates from part 4 compare with the flow rates obtained in Figure 4C of Leong et al.? (0.2 pts)</font>**
- ### **<font color="red"> How do they compare to the theoretical flow rates you calculated in Part 3? (0.2 pts)</font>**
- ### **<font color="red">If your chip failed, note any modes of failure. (0.2 pts)</font>**
RBV not strong enough

### TO-DO: The measured flow rates from part 4 are much larger than the flow rates obtained in Figure 4C but follow the same pattern where their measured values for flow rate vary from the theoretical but slowly become more accurate downstream. The only outlier is reservoir 4 which has a dramatic drop that is half of the theoretical.

### In comparison to the theoretical flow rates calculated in part 3, Reservoir 1 and 4 are the farthest while reservoirs 2 and 3 stay within a tolerance of plus/minus 0.7. 

### In my chip, reservoir 4 started draining the moment reservoir 1 started, going down in increments after each reservoir upstream finishes. This could be due to a weak RBV valve where the geometry is off, causing it to have a smaller capillary pressure barrier than what is intended. This means at each moment the pressure changed from the other reservoirs emptying, RBV 4 capillary pressure dropped below the pumps pressure, causing it to leak. 